src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationAbstractNamedTypeSymbol.cs

```diff
@@ -53,13 +54,13 @@ public INamedTypeSymbol Construct(params ITypeSymbol[] typeArguments)
             }
 
             return new CodeGenerationConstructedNamedTypeSymbol(
-                this, typeArguments.ToImmutableArray(), this.TypeMembers);
+                (CodeGenerationNamedTypeSymbol)this.ConstructedFrom, typeArguments.ToImmutableArray(), this.TypeMembers);
```

Comment:
💡 Consider making `ConstructedFrom` return `CodeGenerationNamedTypeSymbol`, and provide an explicit interface implementation that returns `INamedTypeSymbol`.

Reply:
I thought about it, decided meh, and now you're making me feel bad. Fixing.